kasa/iot/modules/emeter.py

```diff
@@ -6,128 +6,71 @@
-                container=self,
-                unit="V",
-                id="voltage",  # for homeassistant backwards compat
-                precision_hint=1,
-                category=Feature.Category.Primary,
-            )
-        )
-        self._add_feature(
-            Feature(
-                device,
-                name="Current",
-                attribute_getter="current",
-                container=self,
-                unit="A",
-                id="current_a",  # for homeassistant backwards compat
-                precision_hint=2,
-                category=Feature.Category.Primary,
-            )
-        )
 
     @property  # type: ignore
-    def realtime(self) -> EmeterStatus:
+    def status(self) -> EmeterStatus:
         """Return current energy readings."""
         return EmeterStatus(self.data["get_realtime"])
 
     @property
-    def emeter_today(self) -> float | None:
+    def consumption_today(self) -> float | None:
         """Return today's energy consumption in kWh."""
         raw_data = self.daily_data
         today = datetime.now().day
         data = self._convert_stat_data(raw_data, entry_key="day", key=today)
-        return data.get(today)
+        return data.get(today, 0.0)
 
     @property
-    def emeter_this_month(self) -> float | None:
+    def consumption_this_month(self) -> float | None:
         """Return this month's energy consumption in kWh."""
         raw_data = self.monthly_data
         current_month = datetime.now().month
         data = self._convert_stat_data(raw_data, entry_key="month", key=current_month)
-        return data.get(current_month)
+        return data.get(current_month, 0.0)
 
     @property
     def current_consumption(self) -> float | None:
         """Get the current power consumption in Watt."""
-        return self.realtime.power
+        return self.status.power
 
     @property
-    def emeter_total(self) -> float | None:
+    def consumption_total(self) -> float | None:
         """Return total consumption since last reboot in kWh."""
-        return self.realtime.total
+        return self.status.total
 
     @property
     def current(self) -> float | None:
         """Return the current in A."""
-        return self.realtime.current
+        return self.status.current
 
     @property
     def voltage(self) -> float | None:
         """Get the current voltage in V."""
-        return self.realtime.voltage
+        return self.status.voltage
+
+    @property
+    def has_voltage_current(self) -> bool:
+        """Return True if the device reports current and voltage."""
+        return True
+
+    @property
+    def has_total_consumption(self) -> bool:
+        """Return True if device reports total energy consumption since last reboot."""
+        return True
+
+    @property
+    def has_periodic_stats(self) -> bool:
+        """Return True if device can report statistics for different time periods."""
+        return True
```

Comment:
I'd rather avoid having these at all, if possible.

Reply:
As per comment above I think we should keep

Reply:
Replace with `supports` `ModuleFeature` flags